ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -376,7 +399,7 @@ const NetworksForm = ({
         warningMessage = t('failedToFetchTickerSymbolData');
       } else {
         const matchedChain = safeChainsList?.find(
-          (chain) => chain.chainId.toString() === formChainId,
+          (chain) => `0x${chain.chainId.toString(16)}` === hexFormChainId,
```

Comment:
This this line too presumptuous?  If a user enters a hex chain ID in the form originally, would this `0x${chain.chainId.toString(16)}` part break?

Reply:
The chain.chainId part is coming from the safeChainsList > chainid.network fetch call. I don't think this would break unless they changed the way chainIds were being returned from their API. I can probably use our toHex function here still though, so thanks.

Reply:
removed in [1a17e54](https://github.com/MetaMask/metamask-extension/pull/21660/commits/1a17e54c8fd704b58a6c0301a3a6ffa19ba29379)
Turns out since resetForm isn't being called continuously anymore, we no longer need to convert here